tests/testthat/test-generate_data.R

```diff
@@ -381,3 +381,176 @@ test_that("genDummy works.", {
+  expect_error(genMarkov(n=10, transMat = mat6, chainLen = 5, wide = TRUE, startProb = ".7;.3"), class = "simstudy::lengthMismatch")
+  
+  set.seed(oldSeed)
+  
+})
+
+test_that("genMarkov works.", {
+  oldSeed <- .Random.seed
+  set.seed(23456)
+  
+  
+  # not startProb
+  ## pk
+  mat_pow <- function(x, k) {
+    x.k <- x
+    for (i in 2:k) {
+      x.k <- x.k %*% x
+    }
+    x.k
+  }
+  
+  matr <- t(matrix(c(0.5, 0.5, 0.0, 0.0,
+                     0.15, 0.5, 0.35, 0.0,
+                     0.0, 0.35, 0.5, 0.15,
+                     0.0, 0.0, 0.5, 0.5), nrow=4,ncol=4))
+  
+  theoretical_p <- mat_pow(matr, 100)[1,]
+  
+  nind <- 10
+  nchain <- 5000
+  
+  gm1 <- genMarkov(n = nind, transMat = matr, chainLen = nchain)
+  
+  prop <- gm1[, .N, keyby = .(id,state)]
+  prop[, p := N/sum(N), keyby = id]
+  prop[, t.p := rep(theoretical_p, max(id))]
+  prop[, dif := abs(p - t.p)]
+  
+  expect_true(all(prop[, dif < 0.02]))
+  
+  ## correct number of events gen
+  expect_equal(nchain, gm1[, max(period)])
+  
+  ## number of categories == dimensions of transistion matrix
+  expect_true((length(table(gm1$state)) == dim(matr)[1]) & (gm1[, max(state)] == dim(matr)[1]))
+  
+  # startProb
+  ## pk
+  set.seed(23456)
+  gm2 <- genMarkov(n=nind, transMat = matr, chainLen = nchain, startProb = "0.65;0.25;0.05;0.05")
+  
+  prop <- gm1[, .N, keyby = .(id,state)]
+  prop[, p := N/sum(N), keyby = id]
+  prop[, t.p := rep(theoretical_p, max(id))]
+  prop[, dif := abs(p - t.p)]
+  
+  expect_true(all(prop[, dif < 0.02]))
+  
+  ## correct number of events gen
+  expect_equal(nchain, gm2[, max(period)])
+  
+  ## number of categories == dimensions of transistion matrix
+  expect_true((length(table(gm2$state)) == dim(matr)[1]) & (gm2[, max(state)] == dim(matr)[1]))
+  
+  
+  # not wide == wide
+  set.seed(23456)
+  gm1_w <- genMarkov(n=nind, transMat = matr, chainLen = nchain, wide = TRUE)
+  
+  set.seed(23456)
+  gm2_w <- genMarkov(n=nind, transMat = matr, chainLen = nchain, wide = TRUE, startProb = "0.65;0.25;0.05;0.05")
+  
+  check_equal <- function (gm_not_wide, gm_wide) {
+    rand_id <- sample(nind, 1)
+    rand_state <- sample(nchain, 1)
+    gmnw <- gm_not_wide[id == rand_id & period == rand_state, state]
+    gmw <- gm_wide[rand_id, rand_state + 1, with = FALSE]
+    
+    expect_equal(as.numeric(gmnw), as.numeric(gmw))
+  }
+  
+  for (i in 1:5) {
+    check_equal(gm1, gm1_w)
+    check_equal(gm2, gm2_w)
+  }
+  
+  set.seed(oldSeed)
+  
+})
+
+#genMultiFac ----
+test_that("genMultiFac throws errors.", {
+  oldSeed <- .Random.seed
+  set.seed(98765)
+  
+  # check nFactors are integers
+  expect_error(genMultiFac(1.4, each = 4), class = "simstudy::wrongType")
+  
+  # check length nFactors greater than 2
+  expect_error(genMultiFac(1, each = 4), class = "simstudy::greaterThan")
+  
+  # check number of levels matches factors
+  expect_error(genMultiFac(3, levels = c(2, 3)), class = "simstudy::notEqual")
+  
+  set.seed(oldSeed)
```

Comment:
not all of these tests works and I'm not sure why. They pass just find when run in the console and have all the expected outputs, but when I run them as a batch with all the other tests in the file, they give an error and a warning

Reply:
I'll check it out and see if I can uncover anything.